flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/JsonRowDeserializationSchema.java

```diff
@@ -242,11 +247,25 @@ private DeserializationRuntimeConverter wrapIntoNullableConverter(Deserializatio
 			return Optional.of(createObjectArrayConverter(((BasicArrayTypeInfo) typeInfo).getComponentInfo()));
 		} else if (isPrimitiveByteArray(typeInfo)) {
 			return Optional.of(createByteArrayConverter());
+		} else if (typeInfo instanceof MapTypeInfo) {
+			MapTypeInfo<?, ?> mapTypeInfo = (MapTypeInfo<?, ?>) typeInfo;
+			return Optional.of(createMapConverter(mapTypeInfo.getKeyTypeInfo(), mapTypeInfo.getValueTypeInfo()));
 		} else {
 			return Optional.empty();
 		}
 	}
 
+	private DeserializationRuntimeConverter createMapConverter(TypeInformation keyType, TypeInformation valueType) {
+		DeserializationRuntimeConverter valueConverter = createConverter(valueType);
+		DeserializationRuntimeConverter keyConverter = createConverter(keyType);
+
+		return (mapper, jsonNode) -> StreamSupport.stream(
```

Comment:
According to [flink code style](https://flink.apache.org/contributing/code-style-and-quality-java.html), we should avoid Java Streams in any performance critical code. Since this DeserializationRuntimeConverter will be called in per record fashion, I would suggest to write it without Java Streams. 
(I know the code codes are written with Java Streams, but we should follow the code style in newly added codes)

Reply:
Yes, it's called per record. I'll change it back according to the style guide.

Reply:
done.